scripts/linters/pylint_extensions.py

```diff
@@ -165,6 +174,24 @@ def visit_call(self, node):
                             func_name))
                     num_positional_args_unused -= 1
 
+        for arg in keyword_args:
+            # If there is *args and **kwargs in the functin definition skip the
+            # check because we can use keywords arguments in function call even
+            # if **kwargs is present in the function definition.
+            if not called.args.kwarg and callable_name != 'constructor':
```

Comment:
I do not get why we are by passing function with **kwargs? 

Also, this check is just checking for `kwargs` and not when the argument is `args` as mentioned in the comment, right?

Reply:
Because if there is  *args or **kwargs in the function definition, then we can use keyword arguments in the function call even if we do not have any keyword argument in the function definition. For example
```
def func(*args, **kwargs):


func(a=1, b=2, c=3)
```

Reply:
So, should be make this check this broad? Can't it be a specific? For example functions like:

```
def func(entity_id, *args, **kwargs):

func(entity_id='1', a=1, b=2, c=3)

```

Shouldn't we remove `entity_id=` from such functions?

Reply:
Yes, we should, but there is not a way to identify those, because it will all be mixed. Following your example here.
by parsing calling method we get keyword_arguments = entity_id, a, b, c.
But from the function call, we will get keyword_arguments = []
now we do not have a way to identify which one is a keyword argument and which one is not.

Reply:
I see, got it.

Reply:
I get it. Don't you think a contributor would not understand much reading your comment? 

May be extend it a bit and include an example?

Reply:
Done.